GameOfLife/GameOfLife/Board/Board.cs

```diff
@@ -0,0 +1,79 @@
+﻿
+namespace GameOfLife
+{
+    /// <summary>
+    /// Class for storing and managing field and cells.
+    /// </summary>
+    [Serializable]
+    public abstract class Board : IBoard
+    {
+        /// <summary>
+        /// Unique identifier of the board.
+        /// </summary>10
+        public Guid BoardId { get; } = Guid.NewGuid();
+
+        /// <summary>
+        /// Holds statistics about the game.
+        /// </summary>
+        public Dictionary<string, int> Statistics { get; set; } = new Dictionary<string, int>();
+
+        /// <summary>
+        /// Two-dimensional array in which initial values are stored and new layout created.
+        /// </summary>
+        public bool[,] InitialBoard { get; set; }
+
+        /// <summary>
+        /// Characters that indicates live cell on a board.
+        /// </summary>
+        protected const char LiveCell = (char)009632;
+
+        /// <summary>
+        /// Used as a space between cells if used once, and dead cell if used twice.
+        /// </summary>
+        protected const string EmptyCell = " ";
+
+        /// <summary>
+        /// Constructor creates two-dimensional array with user-desired sizes
+        /// and populates it with randomly generated boolean values.
+        /// </summary>
+        /// <param name="height">Number of rows stacked vertically.</param>
+        /// <param name="width">Number of values that can be stored in a row.</param>
+        public Board(int height, int width)
+        {
+            InitialBoard = new bool[height, width];
+        }
+        
+        /// <summary>
+        /// Constructor for testing purposes.
+        /// </summary>
+        /// <param name="layout">Array contains predefined boolean values of the field.</param>
+        public Board(bool[,] layout)
+        {
+            InitialBoard = layout;
+        }
+
+        /// <summary>
+        /// Parameterless constructor that turned out to be very crucial.
+        /// </summary>
+        public Board()
```

Comment:
Why this is crucial? I think c# handles situations when class is created without any parameters without this empty constructor.

Reply:
It wasn't crucial at all. After refactoring code and adding this constructor, the code coincidentally started working, so I assumed it was due to the constructor.